webapp/channels/src/components/post_edit_history/post_edit_history.tsx

```diff
@@ -37,25 +42,67 @@ const renderThumbVertical = (props: Record<string, unknown>): JSX.Element => (
 const PostEditHistory = ({
     channelDisplayName,
     originalPost,
-    postEditHistory,
 }: PropsFromRedux) => {
+    const [postEditHistory, setPostEditHistory] = useState<Post[]>([]);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [isLoading, setIsLoading] = useState<boolean>(false);
     const scrollbars = useRef<Scrollbars | null>(null);
     const {formatMessage} = useIntl();
+    const retrieveErrorHeading = formatMessage({
+        id: 'post_info.edit.history.retrieveError',
+        defaultMessage: 'Unable to load edit history',
+    });
+    const retrieveErrorSubheading = formatMessage({
+        id: 'post_info.edit.history.retrieveErrorVerbose',
+        defaultMessage: 'There was an error loading the history for this message. Check your network connection or try again later.',
+    });
 
     useEffect(() => {
+        const fetchPostEditHistory = async () => {
+            try {
+                setIsLoading(true);
+                const history = await Client4.getPostEditHistory(originalPost.id);
+                setPostEditHistory(history);
+                setHasError(false);
+            } catch (error) {
+                setHasError(true);
+                setPostEditHistory([]);
+            } finally {
+                setIsLoading(false);
+            }
+        };
+        fetchPostEditHistory();
         scrollbars.current?.scrollToTop();
-    }, [originalPost, postEditHistory]);
+    }, [originalPost]);
```

Comment:
Could this be changed to `originalPost.id` to avoid reloading the history if the post changes for some other reason? If ESLint complains about that, you could probably get around it by saving `originalPostId` as a separate variable before using it inside the `useEffect`